### PR TITLE
add support for --import-file when using the Automation API

### PR DESCRIPTION
**Pull request by charlie-haley**
# Description

Fixes # (issue)

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

**Comment by charlie-haley** [comment]
> Thanks for the PR, this looks like a good addition to the automation API.
> 
> Having a look at the code, I think this needs to be a string, to be able to pass an actual path argument to the CLI, I don't think this is just a binary flag.

Ah, good point! The output of the help command is quite confusing as it says `--import-file pulumi import` instead of `--import-file string`, I believe that's happened because of `pulumi import` being in single quotes here:
https://github.com/pulumi/pulumi/blob/master/pkg/cmd/pulumi/preview.go#L542

I'll also update that as part of this PR unless you want me to raise a separate PR, also happy to do that.

File: pkg/cmd/pulumi/preview.go [hunk 1]
```diff
@@ -539,7 +539,7 @@ func newPreviewCmd() *cobra.Command {
 	}
 	cmd.PersistentFlags().StringVar(
 		&importFilePath, "import-file", "",
-		"Save any creates seen during the preview into an import file to use with `pulumi import`")
+		"Save any creates seen during the preview into an import file to use with pulumi import")
```
Comment (tgummerer) [inline pkg/cmd/pulumi/preview.go:542]:
```suggestion
		"Save any creates seen during the preview into an import file to use with 'pulumi import'")
```

Maybe having this in single quotes would help distinguish it a little better as being a shell command?

**Review by tgummerer — APPROVED** [review]
Thank you!  Agreed on the docstring, it's nice to update that so the help output is less confusing. I'm fine with keeping that in the same PR, just left a minor suggestion on formatting there.

**Comment by tgummerer** [comment]
@charlie-haley I saw CI was still failing and had a quick look.  It looks like there was an `!= ""` missing in the Go if statement. I took the liberty to just push that to your branch, which will hopefully get the PR green, so we can go ahead and merge it.

**Comment by charlie-haley** [comment]
> @charlie-haley I saw CI was still failing and had a quick look. It looks like there was an `!= ""` missing in the Go if statement. I took the liberty to just push that to your branch, which will hopefully get the PR green, so we can go ahead and merge it.

Thanks! There was also an indentation problem in the stack.ts, it should hopefully all be green now

**Comment by tgummerer** [comment]
Thanks again for this contribution @charlie-haley!  This is gonna go into the next pulumi release, which might happen this week (or potentially next, since we have a company offsite this week)

**Pull request merged**